### DISCUSSION TITLE
How to kill ongoing tasks/terminals when losing connection?

### DISCUSSION THREAD

**Discussion by huw** [community/community#16178] [Codespaces]
I’ll often lose my connection to my Codespace while working, for example when putting my machine to sleep. Unfortunately, I usually have an ongoing task or terminal process in the background, and when the Codespace UI reloads, it is unable to restore those windows.

I then have to either restart the Codespace, or find the process ID and kill it manually. Both are very annoying, especially since this happens a lot.

I’m assuming it’s too hard to restore the terminals, in which case I’d at least like to try and kill them when I lose my connection to the Codespaces UI. Is this possible?

**Comment by philhagger** [comment]
Did anything happen with this? I have the same issue.

**Reply by huw** [reply]
I haven’t run into this issue in months, tbh
